tutorials/preprocessing/plot_40_artifact_correction_ica.py

```diff
@@ -1,304 +1,493 @@
+#      sensor data
+#
+#
+# See the Notes section of the :class:`~mne.preprocessing.ICA` documentation
+# for further details. Next we'll walk through an extended example that
+# illustrates each of these steps in greater detail.
+#
+#
+# Example: EOG and ECG artifact repair
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# Let's begin by visualizing the artifacts that we want to repair. In this
+# dataset they are big enough to see easily in the raw data:
 
-# we will also set state of the random number generator - ICA is a
-# non-deterministic algorithm, but we want to have the same decomposition
-# and the same order of components each time this tutorial is run
-random_state = 23
+# plot some channels that clearly show heartbeats and blinks
+regexp = r'(MEG [12][45][123]1|EEG 00.)'
+artifact_picks = mne.pick_channels_regexp(raw.ch_names, regexp=regexp)
+raw.plot(order=artifact_picks, n_channels=len(artifact_picks))
 
 ###############################################################################
-# Define the ICA object instance
-ica = ICA(n_components=n_components, method=method, random_state=random_state)
-print(ica)
+# We can get a better view of the EOG artifacts using
+# :func:`~mne.preprocessing.create_eog_epochs` like we did in the
+# :ref:`tut-artifact-overview` tutorial:
 
-###############################################################################
-# we avoid fitting ICA on crazy environmental artifacts that would
-# dominate the variance and decomposition
-reject = dict(mag=5e-12, grad=4000e-13)
-ica.fit(raw, picks='meg', decim=decim, reject=reject)
-print(ica)
+eog_evoked = create_eog_epochs(raw).average()
+eog_evoked.plot_joint()
 
 ###############################################################################
-# Plot ICA components
-ica.plot_components()  # can you spot some potential bad guys?
+# Now we'll do the same for the ECG artifacts, using
+# :func:`~mne.preprocessing.create_ecg_epochs`:
 
+ecg_evoked = create_ecg_epochs(raw).average()
+ecg_evoked.plot_joint()
 
 ###############################################################################
-# Component properties
-# --------------------
-#
-# Let's take a closer look at properties of first three independent components.
+# Before we run the ICA, an important step is filtering the data to remove
+# low-frequency drifts, which can negatively affect the quality of the ICA fit.
+# A 1 Hz cutoff frequency is recommended; we'll keep a copy of the unfiltered
+# :class:`~mne.io.Raw` object around so we can apply the ICA solution to it
+# later.
 
-# first, component 0:
-ica.plot_properties(raw, picks=0)
+filt_raw = raw.copy()
+filt_raw.load_data().filter(l_freq=1., h_freq=None)
 
 ###############################################################################
-# we can see that the data were filtered so the spectrum plot is not
-# very informative, let's change that:
-ica.plot_properties(raw, picks=0, psd_args={'fmax': 35.})
+# .. sidebar:: ICA ignores the time domain
```

Comment:
To be technical, I think SOBI uses time-domain information. Of course, most ICA algos don't.